src/utils/request/request.js

```diff
@@ -10,7 +10,7 @@ const request = async (url, fetchOptions = {}, configuration = {}) => {
     const firstRequestData = await singleRequest(url, { ...fetchOptions, page: { number: 1 } });
 
     if (firstRequestData.meta === undefined || firstRequestData.meta.matches === undefined) {
-      return applyPlugins(firstRequestData, responsePlugins);
+      throw new Error('This endpoint does not have pagination and therefore fetchAll will not work here.');
```

Comment:
Can we also specify the endpoint in this error please? Can come in useful when doing multiple requests.

Reply:
Maybe explicitly say that it's missing the matches data.
Because for example  `invoices.list` has pagination, but just doesn't have the data in the response.

Reply:
IMO it is then a bug of the API endpoint and not a bug of the user of the SDK ... But yeah, I can adapt it.